Makefile

```diff
@@ -552,18 +552,25 @@ ifdef TILES
   endif
 else
   ifeq ($(CROSS),)
-    ifneq ($(shell which ncursesw5-config 2>/dev/null),)
-      HAVE_NCURSESW5CONFIG = 1
+  # Check to see if pkg-config is on the system. 
+    ifneq ($(shell which pkg-config 2>/dev/null),)
+      HAVE_PKGCONFIG = 1
     endif
   endif
 
   # Link to ncurses if we're using a non-tiles, Linux build
-  ifeq ($(HAVE_NCURSESW5CONFIG),1)
-    CXXFLAGS += $(shell ncursesw5-config --cflags)
-    LDFLAGS += $(shell ncursesw5-config --libs)
+  ifeq ($(HAVE_PKGCONFIG),1)
+    CXXFLAGS += $(shell pkg-config --cflags ncurses)
+    LDFLAGS += $(shell pkg-config --libs ncurses)
   else
     ifneq ($(TARGETSYSTEM),WINDOWS)
-      LDFLAGS += -lncurses
+  # This is here for backward comaptablity as a fallback. 
+      ifneq ($(shell which ncurses5-config 2>/dev/null),)
```

Comment:
This fallback doesn't have the same semantics as the original code that was removed, which was inside a further `ifeq ($(CROSS),)` (i.e., "if `$(CROSS)` is unset").

I rather imagine you should've kept the `HAVE_NCURSESW5CONFIG` variable and kept the check in the same place it was before -- only prefixing the check for `pkg-config`. I think this is actually what caused the error in #22435.

Reply:
Yes it does. The code for doing anything with ncurses only runs if your not doing cross. All that does there is check to see if cross is being used and if not set the flag. This also would not have caused the error in that as this was only merged in 17 hours ago as of my posting.

Reply:
17 hours ago, a push to master by kevingranade contained this PR, as well as #22423, which fixed the previous error with the OSX build, reported at #22421.

If you follow the logic in the Makefile, the blocks are (in some kind of pseudocode):
```
if TILES is set {
    // stuff that wasn't changed
} else {
    if CROSS is unset {
        if $(shell which pkg-config) is not empty {
            HAVE_PKGCONFIG = 1
       }
    }

    if HAVE_PKGCONFIG == 1 {
        // ...use pkg-config...
    } else {
        if TARGETSYSTEM is not "WINDOWS" {
            if $(shell which ncurses5-config) is not empty {
                // ...use ncurses-config...
            } else {
                // add "-lncurses" to LDFLAGS
            }
        }

        if OSXCROSS is defined {
            // ... add references to $(LIBSDIR)/ncurses/{lib,include} to {LD,CXX}FLAGS
        }
    }
}
```